src/components/policy/policy_external/src/policy_helper.cc

```diff
@@ -620,33 +620,50 @@ void FillNotificationData::UpdateParameters(
   ParametersConstItr it_parameters = in_parameters.begin();
   ParametersConstItr it_parameters_end = in_parameters.end();
 
-  // Due to APPLINK-24201 SDL must consider cases when 'parameters' section is
-  // not present for RPC or present, but is empty.
-
-  // If 'parameters' section is like: 'parameters' : []
-  if (in_parameters.is_initialized() && in_parameters.empty()) {
-    if (!does_require_user_consent_) {
-      out_parameter.any_parameter_disallowed_by_policy = true;
-    }
-    if (does_require_user_consent_ && kAllowedKey == current_key_) {
-      out_parameter.any_parameter_disallowed_by_user = true;
+  // From AppLink Policies Manager specification:
+  // To determine consent for a particular RPC in a particular HMI level with
+  // particular parameters (if applicable), the system shall find all of the
+  // functional groups the RPC is included in. If user consent is needed as
+  // listed within the functional group in the policy table, the system shall
+  // use a logical AND: backend permissions AND User permissions. If the RPC is
+  // listed under more than one group, the system shall perform a logical OR
+  // amongst all of the possible allowed permissions scenarios for the RPC (and
+  // parameter/or HMI level) defined by each of the functional groups.
+
+  if (!IsSomeParametersAllowed(out_parameter)) {
```

Comment:
This seems to short-circuit the adding of parameters within it, I think this conditional doesn't need to be here

Reply:
@conlain-k removed in 80291097033c8a6cd44a95f2ff4ff4def57e5e8e